nck/readers/twitter/reader.py

```diff
@@ -311,23 +311,27 @@ def get_campaign_management_report(self):
             for entity_obj in ACCOUNT_CHILD_OBJECTS[self.entity]
         ]
 
+    @retry(stop=stop_after_delay(901))
     def get_cards_report(self):
         """
         Get 'ENTITY' report through the 'Creatives' endpoint of Twitter Ads API.
         Supported entities: CARD
         Documentation: https://developer.twitter.com/en/docs/ads/creatives/api-reference/
         """
-
         for tweet in self.get_published_tweets():
-            if "card_uri" in tweet:
-                card_fetch = self.get_card_fetch(card_uri=tweet["card_uri"])
-                card_attributes = {attr: getattr(card_fetch, attr, None) for attr in self.entity_attributes}
-                record = {
-                    "tweet_id": tweet["tweet_id"],
-                    "card_uri": tweet["card_uri"],
-                    **card_attributes,
-                }
-                yield record
+            try:
+                if "card_uri" in tweet:
+                    card_fetch = self.get_card_fetch(card_uri=tweet["card_uri"])
+                    card_attributes = {attr: getattr(card_fetch, attr, None) for attr in self.entity_attributes}
+                    record = {
+                        "tweet_id": tweet["tweet_id"],
+                        "card_uri": tweet["card_uri"],
+                        **card_attributes,
+                    }
+                    yield record
+            except Exception:
+                ex_type, ex, tb = sys.exc_info()
+                logger.warning(f"Failed to ingest post with error: {ex}. Traceback: {traceback.print_tb(tb)}")
```

Comment:
Is it necessary to get the all the logs and traceback here and not only a custom Exception?

Open discussion